### ISSUE TITLE
Rewrite/Merge group-by extension

### ISSUE THREAD

**Issue by UtechtDustin** [wenzhixin/bootstrap-table#4413]
All three group-by extensions (two existing and #3804) become to one extension to prevent multiple plugins with the same name and a similar functionality.

The merge will be done by @djhvscf 

This Issues should be checked before submit a PR:
- [ ] #4139
- [ ] #4043
- [ ] #3897
- [ ] #3893
- [ ] #3528
- [ ] #2922
- [ ] #2689
- [ ] #2257
- [ ] #3091
- [ ] #3948
- [ ] #3853

Optional Feature (can be done later):
- [ ] #4146
- [ ] #2148

**Comment by djhvscf** [comment]
Thanks @UtechtDustin

**Comment by wenzhixin** [comment]
The related issues are already closed. I think we can close this issue.

**Comment by UtechtDustin** [comment]
I don't think this issue can be closed, its a collection of issues for the rewrite for @djhvscf.
As you can see in the linked issues there was closed on the same day.

**Comment by wenzhixin** [comment]
OK

**Comment by djhvscf** [comment]
Yes.. this is still in my queue! Sorry

**Comment by muratcakmaksoftware** [comment]
That is big problem what we can do now ?
Firstly We need to still server-side work correctly
A temporary correction can be made.
